src/libsystemd-network/sd-dhcp-server.c

```diff
@@ -1061,6 +1061,39 @@ static bool address_available(sd_dhcp_server *server, be32_t address) {
         return true;
 }
 
+static int server_get_static_lease(sd_dhcp_server *server, const DHCPRequest *req, DHCPLease **ret) {
+        DHCPLease *static_lease;
+        _cleanup_free_ uint8_t *data = NULL;
+
+        static_lease = hashmap_get(server->static_leases_by_client_id, &req->client_id);
+        if (static_lease) {
+                *ret = TAKE_PTR(static_lease);
+                return 0;
+        }
+
+        /* when no lease is found based on the client id fall back to chaddr */
+        data = new(uint8_t, req->message->hlen + 1);
+        if (!data)
+                return -ENOMEM;
+
+        /* set client id type to 1: Ethernet Link-Layer (RFC 2132) */
+        data[0] = 0x01;
+        memcpy(data + 1, req->message->chaddr, req->message->hlen);
+
+        DHCPClientId c;
+
+        c = (DHCPClientId) {
+                .length = req->message->hlen + 1,
+                .data = data,
+        };
+
+        static_lease = hashmap_get(server->static_leases_by_client_id, &c);
```

Comment:
```c
        static_lease = hashmap_get(server->static_leases_by_client_id,
                                   &(DHCPClientId) {
                                           .length = req->message->hlen + 1,
                                           .data = data,
                                   });
```

Reply:
Updated, hope the indentation is correct (aligned with the first character after the `(` and then indenting with the default of 8 spaces?)